src/JewelryDetail.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
+import request from 'superagent';
+import Spinner from './Spinner.js';
+
+export default class Detail extends Component {
+    state = {
+        jewelry: {},
+    }
+    componentDidMount = async () => {
+
+        this.setState({ loading: true });
+
+        const data = await request.get(`https://katilyn-fullstack-project.herokuapp.com/jewelry=${this.props.match.params.id}`);
+
+        this.setState({
+            loading: false,
+
+            jewelry: data.body.results.find(jewel => jewel.id === this.props.match.params.id),
```

Comment:
For the DetailPage, since you're fetching just for that jewelry id you won't need to find before setting state )